ddtrace/tracer/span.go

```diff
@@ -373,7 +373,7 @@ func (s *span) finish(finishTime int64) {
 				log.Error("Stats channel full, disregarding span.")
 			}
 		}
-		if feats.DropP0s {
+		if feats.Stats && feats.DropP0s {
```

Comment:
Would it make sense to also change `loadAgentFeatures` to only set `DropP0s` if it's also setting `Stats`? e.g.

```diff
diff --git a/ddtrace/tracer/option.go b/ddtrace/tracer/option.go
index 7f61450..269ed8f 100644
--- a/ddtrace/tracer/option.go
+++ b/ddtrace/tracer/option.go
@@ -398,7 +398,6 @@ func (c *config) loadAgentFeatures() {
                log.Error("Decoding features: %v", err)
                return
        }
-       c.agent.DropP0s = info.ClientDropP0s
        c.agent.StatsdPort = info.StatsdPort
        for _, endpoint := range info.Endpoints {
                switch endpoint {
@@ -406,6 +405,7 @@ func (c *config) loadAgentFeatures() {
                        if c.HasFeature("discovery") {
                                // client-stats computation is off by default
                                c.agent.Stats = true
+                               c.agent.DropP0s = info.ClientDropP0s
                        }
                }
        }
```

That might reduce the risk of introducing another bug in the future if `DropP0s` is checked somewhere else without also checking `Stats`.

Reply:
That's a good idea. I feel a little weird about changing the values in the agent features struct, since presumably they should be agent features regardless of whether the tracer is configured one way or another. 

Maybe we should have some function that checks both and not do this here. Something like:
```
func canCalculateStats(...) bool {
    c.HasFeature("discovery") && c.agent.Stats == true
}
func canDropP0s(...) bool {
    c.HasFeature("discovery") && c.agent.Stats == true && c.agent.DropP0s == true
}
```

Reply:
Yeah, adding the helper functions sounds better to me.